ui/components/NodeInputSubmit.tsx

```diff
@@ -77,14 +83,42 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
         <Button
           appearance="link"
           type="button"
-          onClick={() =>
-            (window.location.href =
-              window.location.href + "&use_backup_code=true")
-          }
+          onClick={() => {
+            const newQuery = { ...router.query, ["use_backup_code"]: "true" };
+            void router.push(
+              {
+                pathname: router.pathname,
+                query: newQuery,
+              },
+              undefined,
+              { shallow: true },
+            );
+          }}
         >
           Use backup code instead
         </Button>
       )}
+      {showTotpLink && (
+        <Button
+          appearance="link"
+          tabIndex={5}
+          type="button"
+          onClick={() => {
+            const newQuery = { ...router.query };
+            delete newQuery.use_backup_code;
+            void router.push(
```

Comment:
Should we use `push` or `replace`?

With `push` this gets pushed in the browser history, this means that when you press back you will be redirected to the previous page.

With `replace`, this does not get pushed in the browser history, which means that if you press back you will go to the actually first page.

i.e. the current behavior is that if you go to the `Verify your identity` page, click on`Use backup code instead` and then click back, you will be redirected to the `Authentication code` page again.

Whereas with `replace` you would go back to the original login page (where you input email and password). 

We could even make it so that you go back to the application page instead. (cc @lukasSerelis what do you think about this?)

Reply:
As a user I'd expect the extra history entry. So going back brings me the state before clicking any button or link. So I think push is the right thing to use here, not replace. I think replace should be used on automatic redirects, that don't require a user interaction.

Reply:
+1 on what @edlerd said. The only weirdness that could occur here is that if a user, for whatever unexplainable reason hits "I want to use a Backup code" and then  "I want to use Authenticator app" back and forth for a a lot of times, it'd be great if this wouldn't create a new history entry for every time they go back and forth. Not sure if this is possible, or how such implementation would look like. 

This is a very much nice to have cause this isn't a real case that people would do, just that having this sorted would make it into a very polished redirection experience.